Remove etcd dependency

iancoffey opened the pull request:
- Removes etcd dependency
- Adds ability to confugure alerts via annotations
- Adds initial helm chart

resolves #22

iancoffey commented:
Left to do
- [ ] ~~update travis env to include dockerhub creds, and update deploy script~~ https://github.com/deis/monitor/issues/40
- [x] push initial images
- [x] test helm installation

mboersma commented on README.md:
````diff
@@ -8,7 +8,21 @@ Deis (pronounced DAY-iss) is an open source PaaS that makes it easy to deploy an
+Install [helm](http://helm.sh) and add the monitor repository to your chart list:
+
+```console
+$ helm repo add monitor https://github.com/monitor/charts
````
This should be `helm repo add monitor https://github.com/deis/monitor` I assume. But will that work without a `monitor` subdirectory and a `Chart.yaml` file?

iancoffey replied:
Yep, right. Good catch. I think this will work now, but will test extensively after merge.

mboersma commented on _scripts/deploy.sh:
```diff
@@ -6,7 +6,7 @@
 export IMAGE_PREFIX=deisci BUILD_TAG=v2-beta
-docker login -e="$DOCKER_EMAIL" -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
-DEIS_REGISTRY='' make -C .. build push
+#docker login -e="$DOCKER_EMAIL" -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
```
The general strategy for other components is to push Docker images to both Docker Hub and quay.io, so we shouldn't comment this out unless there's a good reason.

iancoffey replied:
@mboersma Gotcha. I made https://github.com/deis/monitor/issues/40 for that purpose - I just need help getting the creds for dockerhub and ill update this.

arschles commented on alertmanager_boot.go:
```diff
@@ -0,0 +1,94 @@
+aggregation_rule {
+  repeat_rate_seconds: 3600
+  notification_config_name: "alertmanager_deis"
+}`
```
FYI my preference would be to put this as a file on-disk. Not blocking this PR

arschles commented on alertmanager_boot.go:
```diff
@@ -0,0 +1,94 @@
+}
+
+func writeConfig(alertConfig *AlertConfig) error {
+	tmpl, err := template.New("alertmanager").Funcs(sprig.TxtFuncMap()).Parse(alertTemplate)
```
can you use [`template.Must`](https://godoc.org/text/template#Must) here, and move it into package-global scope so that it's only executed once?

iancoffey committed:
feat(etcd): remove dependency on etcd

- add ability to use annotations for alert configuration
- add slack alert handling as initial option
- remove confd and etcd everywhere
- add alertmanager Go boot and necessary Makefile targets

iancoffey committed:
feat(chart): Add initial helm Chart

iancoffey commented:
Tested, still works. Look ok @arschles ?

arschles commented:
@iancoffey yep, LGTM
